src/satosa/exception.py

```diff
@@ -30,6 +30,39 @@ class SATOSACriticalError(SATOSAError):
     """
     pass
 
+class SATOSAProcessingHaltError(SATOSAError):
+    """
+    SATOSA error that should stop processing
+    """
+    def __init__(self, state, message, *args, **kwargs):
+        """
+        :type state: satosa.state.State
+        :param message: str
+        :param args: any
+        :param kwargs: any
+
+        :param state: Satosa state
+        :param message: A test message
+        :param args: whatever
+        :param kwargs: whatever
+        """
+        super().__init__(message, *args, **kwargs)
+        self._message = "One of the plugins or microservices raised an error. Redirect to an error page"
```

Comment:
This is a bit misleading. If this error concerns microservices only, then it should be caught [here](https://github.com/SUNET/SATOSA/blob/master/src/satosa/base.py#L174) and [here](https://github.com/SUNET/SATOSA/blob/master/src/satosa/base.py#L113), around the calls to `process()`. If this is a generic error, then I think it is placed correctly but, this message should change.